backend/mcp_backend/servers/generic_sharepoint_server_multi_dept.py

```
@@ -0,0 +1,1301 @@
+        import zipfile
+        import xml.etree.ElementTree as ET
+
+        # Parse DOCX as ZIP file and extract text from document.xml
+        try:
+            docx_stream = BytesIO(content)
+            with zipfile.ZipFile(docx_stream, "r") as zip_file:
+                # Read the main document content
+                doc_xml = zip_file.read("word/document.xml")
+
+                # Parse XML and extract text
+                root = ET.fromstring(doc_xml)
```

Ruff (0.14.7)

Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)